feat: Documentation category token

albinAppsmith opened the pull request:
## Description

Added documentation for category token.

Fixes https://github.com/appsmithorg/appsmith/issues/26309

## Type of change

- New feature (non-breaking change which adds functionality)
- Chore (housekeeping or task changes that don't impact user perception)

## How Has This Been Tested?
> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.
> Delete anything that is not important

- Manual on storybook 
- Manual on main repo
- Jest
- Cypress

### Test Plan
> Add Testsmith test cases links that relate to this PR

### Issues raised during DP testing
> Link issues raised during DP testing for better visiblity and tracking (copy link from comments dropped on this PR)


## Checklist:
### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes


### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

tanvibhakta commented on packages/design-system/src/Documentation/components/CodeBlock.tsx:
```diff
@@ -0,0 +1,41 @@
+        toast.show("Copied to clipboard", { kind: "success" });
+      }}
+    >
+      <Tooltip content="Click here to copy" mouseEnterDelay={0.7}>
```
Really good idea! Might be a better design to show a copy button up front instead of on hover (the reader might not know that they _can_ hover, so that affordance is missing), but if you feel like this is not a priority you don't need to pick it up. Do create a ticket for it and mark it as `good first issue` if you do table it, though.

tanvibhakta replied:
Can you create a task for this?

albinAppsmith replied:
https://github.com/appsmithorg/appsmith/issues/26847
Already been created

tanvibhakta commented on packages/design-system/src/Documentation/components/ColorBlock.tsx:
```diff
@@ -0,0 +1,38 @@
+import React from "react";
```
Just curious - why did you roll your own component instead of using [storybook's color palette doc block](https://storybook.js.org/docs/react/api/doc-block-colorpalette)?

albinAppsmith replied:
Was not aware of this... will update this later.

albinAppsmith commented on packages/design-system/src/Documentation/CategoryToken/Introduction.mdx:
````diff
@@ -0,0 +1,47 @@
+
+*```Atom-Category-Component-Variant-Surface-State-PresenationalAttribute```*
+
+<table>
````
Will update this later... I find html table much easier to use here.

tanvibhakta commented on packages/design-system/.storybook/main.js:
```diff
@@ -85,12 +85,13 @@ module.exports = {
+    "../src/Documentation/Colors.mdx",
+    "../src/Documentation/Space.mdx",
+    "../src/Documentation/FAQs.mdx",
+    "../src/Documentation/LeanGuidelines.mdx",
```
This still isn't ordered correctly - 
![Uploading Screenshot 2023-09-04 at 9.56.03 AM.png…]()

tanvibhakta commented:
I believe [this page](https://design-system-git-documentation-category-token-get-appsmith.vercel.app/?path=/docs/docs-design-tokens--documentation#how-to-style) still needs to be updated.

tanvibhakta commented on packages/design-system/src/Documentation/CategoryToken/ResponseTokens.mdx:
```diff
@@ -0,0 +1,101 @@
+
+<Meta title="Docs/Category Token/Response Tokens" name="Response Tokens" />
+
+#### Response
```
I still see the lean guidelines page is up - can you please move the examples of the components to it's various parent pages and delete the lean guidelines page?

albinAppsmith commented:
@tanvibhakta fixed, please re-review.

tanvibhakta commented:
1. Is our description of the component tokens accurate? My understanding is that we use the component tokens when the component has a surface that already has a category token defined but for whatever reason that token doesn't work for the component. This is not the understanding I got from the paragraph

![Screenshot 2023-09-07 at 12 28 39 PM](https://github.com/appsmithorg/design-system/assets/13763558/f2610940-4e62-4d3e-ba6a-905130cfb425)

2. It looks like the hex values are missing from the example table

![Screenshot 2023-09-07 at 12 30 29 PM](https://github.com/appsmithorg/design-system/assets/13763558/6c56ff46-c6ab-4230-8ea3-e6791e20214f)

3. There have been several figma links dropped in the docuemntation that link to an image - can we please embed the image directly instead? The context switching hinders the reader's ability to understand the subject 

![Screenshot 2023-09-07 at 12 40 49 PM](https://github.com/appsmithorg/design-system/assets/13763558/7b1f5243-8ff7-4b47-bcb3-71430614b051)
![Screenshot 2023-09-07 at 12 40 57 PM](https://github.com/appsmithorg/design-system/assets/13763558/677791ac-96d2-4909-b972-848a420c6a8c)
![Screenshot 2023-09-07 at 12 41 04 PM](https://github.com/appsmithorg/design-system/assets/13763558/2587d324-e6ec-4670-8152-65813d752bce)

4. This is a great opportunity to explain component-level tokens and why we are not using the category token here. 

![Screenshot 2023-09-07 at 12 42 53 PM](https://github.com/appsmithorg/design-system/assets/13763558/0dddb7f4-8243-4de5-8090-5aa68ea0d9d4)

albinAppsmith commented:
1. @tanvibhakta This was something you had added. Should have thought about it that time itself. Anyways, I have updated this.
2. There is no need to show the token value here. Removed that column.
3. Again this is something you had added. Not changing this now since it is not in scope of my work.
4. This is already been explained in top and category token page.
